File: common/src/main/java/me/kvalbrus/multibans/common/punishment/punishments/MultiTemporaryBan.java
package me.kvalbrus.multibans.common.punishment.punishments;

import java.util.List;
import me.kvalbrus.multibans.api.punishment.creator.PunishmentCreator;
import me.kvalbrus.multibans.api.punishment.target.PunishmentTarget;
import me.kvalbrus.multibans.api.punishment.PunishmentType;
import me.kvalbrus.multibans.api.punishment.punishments.TemporaryBan;
import me.kvalbrus.multibans.common.managers.PluginManager;
import me.kvalbrus.multibans.common.permissions.Permission;
import me.kvalbrus.multibans.common.punishment.MultiTemporaryPunishment;
import me.kvalbrus.multibans.common.utils.Message;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class MultiTemporaryBan extends MultiTemporaryPunishment implements TemporaryBan {

    public MultiTemporaryBan(@NotNull PluginManager pluginManager,
                             @NotNull String id,
                             @NotNull PunishmentTarget target,
                             @NotNull PunishmentCreator creator,
                             long createdDate,
                             long startedDate,
                             long duration,
                             @Nullable String reason,
                             @Nullable String comment,
                             @Nullable PunishmentCreator cancellationCreator,
                             long cancellationDate,
                             @Nullable String cancellationReason,
                             @NotNull List<String> servers,
                             boolean cancelled) {
        super(pluginManager, PunishmentType.TEMP_BAN, id, target, creator, createdDate, startedDate,
            duration, reason, comment, cancellationCreator, cancellationDate, cancellationReason,
            servers, cancelled);
    }

    @NotNull
    @Override
    public String getActivateMessageForListener() {
        return Message.TEMPBAN_ACTIVATE_LISTEN.getMessage();
    }

    @NotNull
    @Override
    public String getActivateMessageForExecutor() {
        return Message.TEMPBAN_ACTIVATE_EXECUTOR.getMessage();
    }

    @Nullable
    @Override
    public String getActivateMessageForTarget() {
        return null;
    }

    @NotNull
    @Override
    public String getDeactivateMessageForListener() {
        return Message.TEMPBAN_DEACTIVATE_LISTEN.getMessage();
    }

    @NotNull
    @Override
    public String getDeactivateMessageForExecutor() {
        return Message.TEMPBAN_DEACTIVATE_EXECUTOR.getMessage();
    }

    @Nullable
    @Override
    public String getDeactivateMessageForTarget() {
        return null;
    }

    @NotNull
    @Override
    public String getDeleteMessageForListener() {
        return Message.TEMPBAN_DELETE_LISTEN.getMessage();
    }

    @NotNull
    @Override
    public String getDeleteMessageForExecutor() {
        return Message.TEMPBAN_DELETE_EXECUTOR.getMessage();
    }

    @Nullable
    @Override
    public String getDeleteMessageForTarget() {
        return null;
    }

    @NotNull
    @Override
    public  String getReasonChangeMessageForExecutor() {
        return Message.TEMPBAN_REASON_CREATE_CHANGE_EXECUTOR.getMessage();
    }

    @NotNull
    @Override
    public String getReasonChangeMessageForListener() {
        return Message.TEMPBAN_REASON_CREATE_CHANGE_LISTEN.getMessage();
    }

    @NotNull
    @Override
    public String getCommentChangeMessageForExecutor() {
        return Message.TEMPBAN_REASON_CREATE_CHANGE_EXECUTOR.getMessage();
    }

    @NotNull
    @Override
    public String getCommentChangeMessageForListener() {
        return Message.TEMPBAN_COMMENT_CHANGE_EXECUTOR.getMessage();
    }

    @NotNull
    @Override
    public Permission getPermissionForListener() {
        return Permission.PUNISHMENT_TEMPBAN_LISTEN;
    }
}

File: common/src/main/kotlin/me/kvalbrus/multibans/common/punishment/punishments/MultiTemporaryBan.kt
package me.kvalbrus.multibans.common.punishment.punishments

import me.kvalbrus.multibans.api.punishment.action.Action
import me.kvalbrus.multibans.api.punishment.action.CreationAction
import me.kvalbrus.multibans.api.punishment.punishments.PunishmentType
import me.kvalbrus.multibans.api.punishment.punishments.TemporaryBan
import me.kvalbrus.multibans.common.managers.PluginManager
import me.kvalbrus.multibans.common.permissions.Permission
import me.kvalbrus.multibans.common.utils.Message.*
import me.kvalbrus.multibans.common.punishment.MultiTemporaryPunishment
import me.kvalbrus.multibans.common.utils.Message

class MultiTemporaryBan : MultiTemporaryPunishment, TemporaryBan {

    constructor(
        pluginManager: PluginManager,
        id: String,
        creationAction: CreationAction,
        activations: MutableList<Action> = mutableListOf(),
        deactivations: MutableList<Action> = mutableListOf(),
        startedDate: Long,
        duration: Long,
        comment: String,
        servers: List<String>,
        cancelled: Boolean) : super(pluginManager, PunishmentType.TEMP_BAN, id, creationAction,
        activations, deactivations, startedDate, duration, comment, servers, cancelled)

    override val activateMessageForListener: Message = TEMPBAN_ACTIVATE_LISTEN
    override val activateMessageForExecutor: Message = TEMPBAN_ACTIVATE_EXECUTOR
    override val activateMessageForTarget: Message = TEMPBAN_ACTIVATE_TARGET
    override val deactivateMessageForListener: Message = TEMPBAN_DEACTIVATE_LISTEN
    override val deactivateMessageForExecutor: Message = TEMPBAN_DEACTIVATE_EXECUTOR
    override val deactivateMessageForTarget: Message = EMPTY
    override val createMessageForListener: Message = TEMPBAN_ACTIVATE_LISTEN
    override val createMessageForExecutor: Message = TEMPBAN_ACTIVATE_EXECUTOR
    override val createMessageForTarget: Message = TEMPBAN_ACTIVATE_TARGET
    override val deleteMessageForListener: Message = TEMPBAN_DELETE_LISTEN
    override val deleteMessageForExecutor: Message = TEMPBAN_DELETE_EXECUTOR
    override val deleteMessageForTarget: Message = TEMPBAN_DELETE_TARGET
    override val reasonChangeMessageForExecutor: Message = TEMPBAN_REASON_CREATE_CHANGE_EXECUTOR
    override val reasonChangeMessageForListener: Message = TEMPBAN_REASON_CREATE_CHANGE_LISTEN
    override val commentChangeMessageForExecutor: Message = TEMPBAN_COMMENT_CHANGE_EXECUTOR
    override val commentChangeMessageForListener: Message = TEMPBAN_COMMENT_CHANGE_LISTEN
    override val permissionForListener: Permission = Permission.PUNISHMENT_TEMPBAN_LISTEN
}